### PR TITLE
Owen/43 implement prof endpoints

### LINKED ISSUE #43: Implement /professor POST and GET endpoints
For the demo we want the ability to add a prof to the DB then retrieve them from the DB using our API.

### PR DESCRIPTION
**Pull request by owenjaques**
# Description
This PR implements the /professor GET and POST endpoints. I had a few more difficulties doing this PR then I thought I would as logging into the database randomly stopped working, so I had to fix that first. Otherwise definitely learned a lot about accessing the database from Python.
  
Fixes #43 

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
- [x] Tried the two endpoints, one with a curl command one from browser. Seems to be working well. 
- [x] Started the entire app with docker-compose and ran through something like what I imagine our demo to be. 
# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have linted my code and fixed all issues

### CODE REVIEW

**Comment by owenjaques** [comment]
I'm noticing this is breaking unit tests but that might take some work to get back as our tests do not (as of yet) start the database. Might be worth fixing in another issue as that seems like a lot of work.

**Review by itsazoo — CHANGES_REQUESTED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -15,4 +15,5 @@ services:
     build:
       context: ./database
       dockerfile: ./Dockerfile
-
+    ports:
+    - 3306:3306
```
[the remark this answers is not in the document]
Comment (owenjaques) [reply docker-compose.yml:19]: We can't do this. These ports are needed to successfully connect to the database from Python. This fixes that bug I mentioned at the top of the PR. Can you please try fully shutting down Docker and rebuilding?

[the remark this answers is not in the document]
Comment (itsazoo) [reply docker-compose.yml:19]: Ah I see, I did do a full restart but I'll give that a shot again

[the remark this answers is not in the document]
Comment (itsazoo) [reply docker-compose.yml:19]: My machine doesn't seem to like 3306:3306, it worked out for 3306. what's the difference between the two?

[the remark this answers is not in the document]
Comment (itsazoo) [reply docker-compose.yml:19]: Oh wait it's probably port forwarding syntax. what's the src and dest of the forwarding?

[the remark this answers is not in the document]
Comment (owenjaques) [reply docker-compose.yml:19]: Hmm, the binding should just ensure that mysql works on 3306 all the time compared to without where its implicitly running on 3306. I don't think there is much difference between them except with 3306:3306 fixed a bug I was having.

**Pull request merged**